### DISCUSSION TITLE
Unable to figure out the node version next.js in my system

### DISCUSSION THREAD

**Discussion by khuranajordan** [vercel/next.js#59518] [Help]
### Summary

```
PS Desktop\test-app> nvm list

  * 20.10.0 (Currently using 64-bit executable)
    18.19.0
    16.20.2
PS Desktop\test-app> npm start

> test-app@0.1.0 start
> next start

You are using Node.js 16.16.0. For Next.js, Node.js version >= v18.17.0 is required.
PS Desktop\test-app> nvm current
v20.10.0
```

Here is a snippet of my terminal. I don't have node 16.16.0 installed but still I am getting this error that my node version is 16.16.0. How to should I resolve this issue?

**Comment by abjutus** [comment]
@khuranajordan
Did you use `v14.0.4` and run scripts with `pnpm (run) xxx`?
I had encountered this issue too.
This should be caused by nodejs version check (#58958) added to `v14.0.4`.

However, running scripts with `npm` did not raise the error, it might caused by `pnpm` only.

After digging into it, you can check `process.versions.node` by running `pnpm version`, you'll find that the version of node you use is different from the version detected by `pnpm`.

So this error will be raised if you had multiple node versions.

For example, with node v21.4.0 installed with nvm

Run `pnpm version`
```
{
  npm: '10.2.3',
  node: '20.10.0',
  ...
}
```

Run `npm version`
```
{
  npm: '10.2.4',
  node: '21.4.0',
  ...
}
```

**Reply by khuranajordan** [reply]
I didn't use pnpm. Instead I installed `yarn` and did `yarn dev`. It runs now but still shows the error if I run it with `npm run dev`.

**Reply by GalMiles** [reply]
I have this issue too!
It works with 'yarn dev' but no with 'npm run dev'. Did you solve this?

**Reply by Luciensday** [reply]
Thank you [khuranajordan](https://github.com/khuranajordan). It works with `yarn dev`

**Reply by chirayudev99** [reply]
facing the same issue, and its working with yarn dev
but we need to find solution for npm run dev

**Comment by Suleman615** [comment]
I was facing same issue , reinstalled node using installer   and after system restart it is working

**Comment by arsheelsheikh** [comment]
upgrade your Next.js form 12 to 13 
run this cmd - 

> npm i next@13 react@latest react-dom@latest eslint-config-next@13

for more refer this- https://nextjs.org/docs/pages/building-your-application/upgrading/version-13

**Comment by francisaguilar21** [comment]
I still experienced this issue with NextJS 14.0.4. I decided to downgrade to NextJS 14.0.3 and it worked perfectly.

Dev Environment Info:
``` bash
{
     npm: '10.2.3',
     node: '20.10.0',
     ...
     next: '14.0.3'
}
```

Hope it works on your end! @Suleman615 @khuranajordan

**Reply by francisaguilar21** [reply]
I just wanted to comment that in `Next 14.1.0`, I still experience this issue with npm. But when I switch to `yarn 1.22.21`, I don't experience this issue.

# Running `npm run build`
![Screenshot 2024-01-25 at 11 33 00 PM](https://github.com/vercel/next.js/assets/25190540/0cc7cdeb-d11a-40b9-b1b2-37314018f988)

# Running `yarn run build`
![Screenshot 2024-01-25 at 11 32 38 PM](https://github.com/vercel/next.js/assets/25190540/ce7b0c4d-bf8f-49a2-9215-47609dd50b89)

I thought to just put it here for anyone who might come across this issue.

**Reply by DanBan451** [reply]
Yes you are correct. I downgraded to next version 14.0.3 and now is working. Thanks!

**Comment by jihostudy** [comment]
I faced this same issue.

I Solved it by Re-installing the latest version of Node.JS

Go to Node.js Website and install the LTS version of NodeJS Again!

I guess the NextJS version was made on the latest version of nodejs??

**Comment by jinyounghwa** [comment]
I was in the same situation. I solved it by installing pnpm. It seems to be an npm issue.
